core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -150,6 +153,7 @@ static void prepareHeaders(Metadata headers, DecompressorRegistry decompressorRe
     if (!advertisedEncodings.isEmpty()) {
       headers.put(MESSAGE_ACCEPT_ENCODING_KEY, advertisedEncodings);
     }
+    statsTraceContext.propagateToHeaders(headers);
```

Comment:
Only some information, e.g., trace ID, some tags, are propagated to the headers. I don't want to give a false impression that it makes a full copy of the contexts.